source/postprocess/volume_of_fluid_statistics.cc

```diff
@@ -0,0 +1,132 @@
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/volume_of_fluid_statistics.h>
+#include <aspect/simulator_access.h>
+#include <aspect/volume_of_fluid/handler.h>
+#include <aspect/global.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    VolumeOfFluidStatistics<dim>::execute (TableHandler &statistics)
+    {
+      AssertThrow(this->get_volume_of_fluid_handler().get_n_fields()!=numbers::invalid_unsigned_int,
+                  ExcMessage("This postprocessor cannot be used without VolumeOfFluid fields"));
+      const QGauss<dim> quadrature_formula (1);
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      unsigned int n_volume_of_fluid_fields = this->get_volume_of_fluid_handler().get_n_fields();
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_quadrature_points |
+                               update_JxW_values);
+      std::vector<double> volume_of_fluid_values(n_q_points);
+
+      std::vector<double> local_volume_of_fluid_vol_sums(n_volume_of_fluid_fields);
+
+      for (unsigned int f=0; f<n_volume_of_fluid_fields; ++f)
+        {
+          FEValuesExtractors::Scalar volume_of_fluid = this->get_volume_of_fluid_handler().field_struct_for_field_index(f)
+                                                       .volume_fraction.extractor_scalar();
+          double volume_of_fluid_vol_sum=0.0, volume_of_fluid_vol_corr=0.0;
+
+          typename DoFHandler<dim>::active_cell_iterator
+          cell = this->get_dof_handler().begin_active(),
+          endc = this->get_dof_handler().end();
+          for (; cell!=endc; ++cell)
+            if (cell->is_locally_owned())
+              {
+                fe_values.reinit (cell);
+                fe_values[volume_of_fluid].get_function_values (this->get_solution(),
+                                                                volume_of_fluid_values);
+                for (unsigned int q = 0; q < n_q_points; ++q)
+                  {
+                    //Use Kahan sum for improved consistency
```

Comment:
Do you have evidence that this is actually necessary, i.e. did you observe problems with summation accuracy? We use simple addition in all of our other statistics postprocessors and I would like to keep the code as simple as possible.

Reply:
Removed as requested.